File: completed/0231.go
func isPowerOfTwo(n int) bool {
    one := false
    for n > 0 {
        if n & 1 != 0 {
            if one {
                return false
            } else {
                one = true
            }
        }
        n >>=1
    }
    return one
}

File: completed/0231.cpp
class Solution {
public:
    bool isPowerOfTwo(int n) {
        if (n == 0) {
            return false;
        }

        while (n > 0) {
            if (n & 1) {
                n >>= 1;
                break;
            }
            n >>= 1;
        }
        return n == 0;
    }
};
